src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts

```diff
@@ -838,7 +838,23 @@ export class SettingsEditor2 extends EditorPane {
 			}
 		}));
 		this._register(this.settingRenderers.onApplyLanguageFilter((lang: string) => {
-			this.focusSearch(`@${LANGUAGE_SETTING_TAG}${lang}`);
+			if (this.searchWidget) {
+				const splitSearchQuery = this.searchWidget.getValue().trimEnd().split(' ');
+				const filter = `@${LANGUAGE_SETTING_TAG}${lang}`;
+
+				// If the query contained the word markdown before,
+				// and we want it to be @lang:markdown, we should just replace
+				// that part of the query directly.
+				const index = splitSearchQuery.findIndex(word => word === lang);
```

Comment:
I'm not sure about this, it's just a guess. I might have searched for a setting with 'markdown' in the name, and now I lost this part of the query. Also I could have typed 'markdo' and gotten to the same place. So I would just add the filter and not worry about trying to guess what the user wants.